src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -112,7 +112,20 @@ def __init__(self, backend, cipher, mode, operation):
             iv_nonce,
             operation,
         )
-        self._backend.openssl_assert(res != 0)
+
+        # Check for XTS mode duplicate keys error
+        errors = self._backend._consume_errors()
+        lib = self._backend._lib
+        if (
+                res == 0
+                and lib.CRYPTOGRAPHY_OPENSSL_111D_OR_GREATER
```

Comment:
If you prefer this test to be `Cryptography_HAS_EVP_R_XTS_DUPLICATED_KEYS`, to future proof this against the same error code being added to LibreSSL, then let me know and I'll change the logic around.